crates/full-node/sov-sequencer/src/preferred/inner.rs

```diff
@@ -265,18 +267,35 @@ where
 
         // DB operations handled by replica-aware db implementation
         let sequence_number = self.get_and_inc_next_sequence_number();
+        let min_profit_per_tx = self.seq_config.sequencer_kind_config.minimum_profit_per_tx;
 
         let start_block_data = StartBlockData {
             sanity_check_visible_slot_number_after_increase: visible_slot_number_after_increase,
             visible_increase,
             node_state_root: node_state_root.clone(),
+            minimum_profit_per_tx: min_profit_per_tx,
         };
 
-        let min_profit_per_tx = self.seq_config.sequencer_kind_config.minimum_profit_per_tx;
+        let old_checkpoint = self
+            .executor
+            .checkpoint
+            .clone_with_empty_witness_dropping_temp_cache();
 
         self.executor
-            .start_rollup_block(start_block_data, min_profit_per_tx)
+            .start_rollup_block(start_block_data.clone())
             .await;
+
+        let state_roots = self.executor.state_roots.clone();
+
+        let notification = StartBtachNotification {
+            state_roots,
```

Comment:
We send the `state_roots` map to the workers, with each worker cloning it in its own thread. The map is small (size ≤5), so the overhead is minimal.

I attempted to send the checkpoint obtained after calling  `start_rollup_block` but ran into panics due to missing `state_roots`. After timeboxing the issue, I chose to send older checkpoint together with the `state_roots`

Mid-term options:
1. Investigate the root cause of the panic, which will likely require changes to `rollup_block_task_body`.
2Use an `Arc` for `state_roots` since workers don’t modify it, though this would require adding custom logic to the execut